### ISSUE TITLE
MainActivity cannot be cast to com.salesforce.androidsdk.phonegap.ui.SalesforceDroidGapActivity

### ISSUE THREAD

**Issue by RianHitori** [forcedotcom/SalesforceMobileSDK-CordovaPlugin#168]
Hi,

I got an issue with the cordova plugin. I successfully install the sample app "contacts app" of salesforce plugin using Android Studio.
After I run the app I got this error:

`getAuthCredentialsError: com.salesforce.contactexplorer.MainActivity cannot be cast to com.salesforce.androidsdk.phonegap.ui.SalesforceDroidGapActivity `

Anyone know about this issue?

Thank you.

**Comment by wmathurin** [comment]
Give us more information.
Did you use forcedroid to create the application?
Was there any error during the application generation?

**Comment by RianHitori** [comment]
Hi @wmathurin,

I followed this instruction - https://developer.salesforce.com/docs/atlas.en-us.mobile_sdk.meta/mobile_sdk/dev_hybrid_apps_run.htm

I ran:
```
git clone https://github.com/forcedotcom/SalesforceMobileSDK-Shared

cordova create contactsApp com.salesforce.contactexplorer contactsApp
cd contactsApp
cordova plugin add https://github.com/forcedotcom/SalesforceMobileSDK-CordovaPlugin
cordova plugin add org.apache.cordova.contacts
cordova plugin add org.apache.cordova.statusbar
cordova platform add android 
cordova platform add ios 
cp -RL <local path to SalesforceMobileSDK-Shared>/samples/contactexplorer/* www/
cordova prepare
```

After that I open it using AndroidStudio. I added 'multiDexEnabled true' on defaultConfig:
```
defaultConfig {
        versionCode cdvVersionCode ?: Integer.parseInt("" + privateHelpers.extractIntFromManifest("versionCode") + "0")
        if (cdvMinSdkVersion != null) {
            minSdkVersion cdvMinSdkVersion
        }
        multiDexEnabled true
    }
```

The build didn't show any error. After I open the app on my android device I see the error above and the salesforce login page is not showing. It works fine on my iOS device.

Thank you.

**Comment by wmathurin** [comment]
I think if you were to do the cordova platform add before you do the cordova plugin add it would work.

Anyway, we just released Mobile SDK 4.1 a few hours ago.
Among other things, we added a "after-platform-add" hook to the plugin, so it should now work regardless of the order of the cordova platform add and cordova plugin add.

Please re-open if you still have issues.

**Comment by RianHitori** [comment]
Thanks @wmathurin. I will test using new version.
